### PR TITLE
fix,enhance,bump(main/luvi): bump to fixed version 2.15.0

### PR DESCRIPTION
**Pull request by mbekkomo**
Bump to fixed version, use shared lpeg, enable zlib, and fix luajit version symbol.

### CODE REVIEW

File: packages/luvi/build.sh [hunk 1]
```diff
@@ -2,36 +2,40 @@ TERMUX_PKG_HOMEPAGE=https://luvit.io
...
 TERMUX_PKG_MAINTAINER="Komo @mbekkomo"
-TERMUX_PKG_VERSION=20241009
-_commit=32b274bed2a291068eda6cf7a296328b86d682a9
-_version=2.14.0+g${_commit::7}
+TERMUX_PKG_VERSION=2.15.0
+TERMUX_PKG_SRCURL=git+https://github.com/luvit/luvi
 TERMUX_PKG_GIT_BRANCH=master
-TERMUX_PKG_SRCURL=git+https://github.com/luvit/luvi.git
-TERMUX_PKG_DEPENDS="pcre2, openssl, luv, libluajit"
+TERMUX_PKG_DEPENDS="pcre2, openssl, luv, lua51-lpeg, zlib, libluajit"
```
Comment (TomJo2000) [inline packages/luvi/build.sh:8]: Please sort these alphabetically.

File: packages/luvi/build.sh [hunk 2]
```diff
@@ -2,36 +2,40 @@ TERMUX_PKG_HOMEPAGE=https://luvit.io
...
+	-DPCRE2_LIBRARIES=${TERMUX_PREFIX}/lib/libpcre2-8.so
+	-DZLIB_LIBRARIES=${TERMUX_PREFIX}/lib/libz.so
+	-DLPEG_LIBRARIES=${TERMUX_PREFIX}/lib/liblpeg-5.1.so
 "
 
-termux_step_post_get_source() {
-	git fetch --unshallow
-	git checkout "${_commit}"
+termux_step_get_source() {
+	git clone --recursive "${TERMUX_PKG_SRCURL/git+/}" "${TERMUX_PKG_SRCDIR}"
```
Comment (TomJo2000) [inline packages/luvi/build.sh:38]: Doesn't the normal `git+` SRCURL code path for getting the source code do a recursive clone?

Comment (mbekkomo) [reply packages/luvi/build.sh:38]: The git command returns command not found error followed by the message unknown `remote-git-https` command or something

Comment (TomJo2000) [reply packages/luvi/build.sh:38]: I'd like to know the exact "or something".

Comment (TomJo2000) [reply packages/luvi/build.sh:38]: The standard `termux_git_clone_src()` code path should do a full incremental recursive clone.
https://github.com/termux/termux-packages/blob/0e462e498b9f6327b748f94a3f5f145ef38c764c/scripts/build/get_source/termux_git_clone_src.sh#L18-L36

**Review by TomJo2000 — APPROVED** [review]

**Pull request merged**